modulus/metrics/general/crps.py

```diff
@@ -50,14 +50,24 @@ def kcrps(pred: Tensor, obs: Tensor, dim: int = 0):
     Tensor
         Map of CRPS
     """
-    pred = pred.unsqueeze(0).transpose(0, dim + 1).squeeze(dim + 1)
-    n = pred.shape[0]
+    n = pred.shape[dim]
+    device = pred.device
     _crps = 0.0 * obs
     for i in range(n):
-        x_i = pred[i]
-        x_j = pred[i:]
-        _crps += torch.abs(x_i - obs) / n
-        _crps -= torch.sum(torch.abs(x_i[None] - x_j) / n, dim=0) / n
+        # x_i = pred[i]
```

Comment:
I didn't review this closely. I suppose again just adding `dim` support?

Reply:
Yes, this is just adding `dim` support.